packages/taco/test/conditions/base/contract.test.ts

```diff
@@ -43,6 +43,11 @@ describe('validation', () => {
       },
     });
   });
+
+  it('infers condition type from constructor', ()=> {
+    const condition = new ContractCondition(testContractConditionObj);
+    expect(condition.value.conditionType).toEqual(ContractConditionType);
+  })
```

Comment:
Seems like `testContractConditionObj` does contain `conditionType` in it, does it also make sense to additionally test removing the value from `testContractConditionObj` and then passing it in? This ensures that the Omit functionality is ok if the `conditionType` isn't provided. Does that make sense or would that situation never happen? If it makes sense, similar comment for other analogous tests that infer type.